content/general/authentication.textile

```diff
@@ -308,10 +352,30 @@ The following Javascript example demonstrates how to issue a token with an expli
   })
 ```
 
-h3(#tokens). Tokens
+h2(#tokens). Tokens
 
-In the documentation, references to tokens typically refer to both @TokenDetails@ object that contain the token string or the token string itself. @TokenDetails@ objects are obtained when "requesting tokens":/realtime/authentication#request-token from the Ably service and contain not only the token string in the @token@ attribute, but also contain attributes describing the properties of the token.
+In the documentation, references to Ably-compatible tokens typically refer either to an Ably Token, or an Ably JWT. For Ably Tokens, this can either be referring to the @TokenDetails@ object that contain the @token@ string or the token string itself. @TokenDetails@ objects are obtained when "requesting an Ably Token":/realtime/authentication#request-token from the Ably service and contain not only the @token@ string in the @token@ attribute, but also contain attributes describing the properties of the Ably Token. For "Ably JWT":#ably-jwt, this will be simply referring to a JWT which has been signed by an Ably private API key.
 
-h4. TokenDetails type
+h3. TokenDetails type
 
 <%= partial 'types/_token_details' %>
+
+h3(#ably-jwt). Ably JWT
+
+An Ably JWT is not strictly an Ably construct, rather it is a "JWT":https://jwt.io/ which has been constructed to be compatible with Ably. The JWT must adhere to the following to ensure compatability:
+
+* *The JOSE header must include:*
+** @kid@ - Key name, such that an API key of @{{API_KEY}}@ will have key name @{{API_KEY_NAME}}@
+* *The JWT claim set must include:*
+** @iat@ - time of issue in seconds
+** @exp@ - expiry time in seconds
+* *The JWT claim set may include:*
+** @x-ably-capability@ - JSON text encoding of the "capability":https://www.ably.io/documentation/general/authentication#tokens
```

Comment:
Does it matter what style of encoding is used i.e. with / without whitespace?  I assume not given we're not constructing doing a checksum in the same way as we do for our tokens.

Reply:
The library canonicalises the JSON in any case so the definition should be the same.